pkg/receive/receivewriter.go

```diff
@@ -0,0 +1,57 @@
+package receive
+
+import (
+	"github.com/go-kit/kit/log"
+	"github.com/improbable-eng/thanos/pkg/store/prompb"
+	"github.com/pkg/errors"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+)
+
+// Appendable returns an Appender.
+type Appendable interface {
+	Appender() (storage.Appender, error)
+}
+
+type ReceiveWriter struct {
```

Comment:
This is spelled `receive.ReceiveWriter` and should be renamed to `receive.Writer` to be more idiomatic.